gordo_components/builder/build_model.py

```diff
@@ -109,7 +109,111 @@ def __init__(
         self.evaluation_config = evaluation_config.copy()
         self.metadata = metadata.copy()
 
-    def build(self) -> Tuple[Optional[sklearn.base.BaseEstimator], dict]:
+    @property
+    def cached_model_path(self) -> Union[os.PathLike, str, None]:
+        return getattr(self, "_cached_model_path", None)
+
+    @cached_model_path.setter
+    def cached_model_path(self, value):
+        self._cached_model_path = value
+
+    def build(
+        self,
+        output_dir: Optional[Union[os.PathLike, str]] = None,
+        model_register_dir: Optional[Union[os.PathLike, str]] = None,
+        replace_cache=False,
+    ) -> Tuple[Optional[sklearn.base.BaseEstimator], dict]:
+        """
+        Build the model and/or ensure that the desired model exists on disk
+        in `output_dir`, and returns the path to it. If `output_dir` exists we
+        assume the model is there (no validation), and return that path.
+
+        If ``output_dir`` is not given, it will simply build the model without
```

Comment:
I think this whole logic was from the time when we copied and returned the path, not the actual models. Today I think it makes sense that the caching can be used without output_dir set (but it wont be able to write to the cache though) .  I think that can make it nicer. Maybe ditch the whole "copy model blabla" code and just try to read from the cache if it is there, and then serialize it to output_dir if that is set (so slightly less effecient since it will load/write instead of copy, but more effeciient)